components/user_settings/sidebar/index.js

```diff
@@ -7,37 +7,41 @@ import {connect} from 'react-redux';
 import {savePreferences} from 'mattermost-redux/actions/preferences';
 import {Preferences} from 'mattermost-redux/constants';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
-import {get as getPreference} from 'mattermost-redux/selectors/entities/preferences';
+import {get as getPreference, getSidebarPreferences} from 'mattermost-redux/selectors/entities/preferences';
 import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
 
-import {GroupUnreadChannels} from 'utils/constants.jsx';
-
 import UserSettingsSidebar from './user_settings_sidebar.jsx';
 
 function mapStateToProps(state) {
     const config = getConfig(state);
 
+    const closeUnusedDirectMessages = getPreference(
+        state,
+        Preferences.CATEGORY_SIDEBAR_SETTINGS,
+        'close_unused_direct_messages',
+        'after_seven_days'
+    );
+
+    const channelSwitcherOption = getPreference(
+        state,
+        Preferences.CATEGORY_SIDEBAR_SETTINGS,
+        'channel_switcher_section',
+        'true'
+    );
+
+    let sidebarPreference = getSidebarPreferences(state);
+    sidebarPreference = {
```

Comment:
This is going to pass a new object each time it's called. Maybe split them out into separate props or just stick with the original names for `unreads_at_top` and `favorite_at_top`?